### PR TITLE
Index service hello world

### PR DESCRIPTION
**Pull request by jjelschen**
Dropwizard project boilerplate and "Hello World" example.

### CODE REVIEW

File: src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldApplication.xtend [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import io.dropwizard.Application
+import io.dropwizard.setup.Bootstrap
+import io.dropwizard.setup.Environment
+import org.testeditor.web.xtext.index.resources.XtextIndexHelloWorldResource
+import org.testeditor.web.xtext.index.health.XtextIndexTemplateHealthCheck
+
+class XtextIndexHelloWorldApplication extends Application<XtextIndexHelloWorldConfiguration> {
+    def static main(String[] args) throws Exception {
+        new XtextIndexHelloWorldApplication().run(args);
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldApplication.xtend]: () ; are relics from the old java world!

File: src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldApplication.xtend [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    override initialize(Bootstrap<XtextIndexHelloWorldConfiguration> bootstrap) {
+        // nothing to do yet
+    }
+
+    override run(XtextIndexHelloWorldConfiguration configuration,
+                    Environment environment) {
+        val resource = new XtextIndexHelloWorldResource(
+        	configuration.template, configuration.defaultName)
+       	val healthCheck = new XtextIndexTemplateHealthCheck(configuration.template)
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldApplication.xtend]: formatting?

File: src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldConfiguration.xtend [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import org.hibernate.validator.constraints.NotEmpty
+
+public class XtextIndexHelloWorldConfiguration extends Configuration {
+    @NotEmpty var String template
+    @NotEmpty var String defaultName = "Stranger"
+
+	new() {
+	}
+
+	@JsonProperty def getTemplate() {
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldConfiguration.xtend]: formatting?

File: src/main/java/org/testeditor/web/xtext/index/health/XtextIndexTemplateHealthCheck.xtend [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    new(String template) {
+        this.template = template
+    }
+
+    override protected Result check() throws Exception {
+        val saying = String.format(template, "TEST")
+        if (!saying.contains("TEST")) {
+            return Result.unhealthy("template doesn't include a name");
+        }
+        return Result.healthy();
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/health/XtextIndexTemplateHealthCheck.xtend]: (); ?

File: src/main/java/org/testeditor/web/xtext/index/health/XtextIndexTemplateHealthCheck.xtend [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    val String template
+
+    new(String template) {
+        this.template = template
+    }
+
+    override protected Result check() throws Exception {
+        val saying = String.format(template, "TEST")
+        if (!saying.contains("TEST")) {
+            return Result.unhealthy("template doesn't include a name");
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/health/XtextIndexTemplateHealthCheck.xtend]: ;

File: src/main/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResource.xtend [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+class XtextIndexHelloWorldResource {
+
+    val String template
+    val String defaultName
+    val AtomicLong counter
+
+    new(String template, String defaultName) {
+        this.template = template
+        this.defaultName = defaultName
+        this.counter = new AtomicLong()
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResource.xtend]: ()

File: src/main/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResource.xtend [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @GET @Timed def sayHello(@QueryParam("name") Optional<String> name) {
+        val value = String.format(template, name.orElse(defaultName))
+        return new Saying(counter.incrementAndGet(), value)
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResource.xtend]: ()

File: src/main/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResource.xtend [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    val String defaultName
+    val AtomicLong counter
+
+    new(String template, String defaultName) {
+        this.template = template
+        this.defaultName = defaultName
+        this.counter = new AtomicLong()
+    }
+
+    @GET @Timed def sayHello(@QueryParam("name") Optional<String> name) {
```
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResource.xtend]: add return type to public methods (even though they are inferrenced)

File: src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend [hunk 9]
```diff
@@ -0,0 +1,38 @@
+package org.testeditor.web.xtext.index.api
+
+import static extension io.dropwizard.testing.FixtureHelpers.*
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend]: organize imports

File: src/test/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResourceTest.xtend [hunk 10]
```diff
@@ -0,0 +1,27 @@
+package org.testeditor.web.xtext.index.resources
+
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/web/xtext/index/resources/XtextIndexHelloWorldResourceTest.xtend]: organize imports

File: src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+import static org.assertj.core.api.Assertions.assertThat
+import io.dropwizard.jackson.Jackson
+import org.junit.Test
+import com.fasterxml.jackson.databind.ObjectMapper
+
+
+class SayingTest {
+	extension ObjectMapper MAPPER = Jackson.newObjectMapper
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend]: type inferrenced?

File: src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+	@Test def void shouldSerializeToJSON() throws Exception {
```
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend]: @Test usually in separate line

File: src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldApplication.xtend (hunk 1 above)
Comment (jjelschen) [reply src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldApplication.xtend]: semicolon is redundant, braces of constructor call are not.

File: src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend (hunk 11 above)
Comment (jjelschen) [reply src/test/java/org/testeditor/web/xtext/index/api/SayingTest.xtend]: Tried that before, won't compile (`ERROR:MAPPER cannot be resolved to a type.`)

File: src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldConfiguration.xtend (hunk 3 above)
Comment (jjelschen) [reply src/main/java/org/testeditor/web/xtext/index/XtextIndexHelloWorldConfiguration.xtend]: ran the auto-formatter… What needs changing? Annotation on its own line…?

**Pull request merged**